src/core/qgsmaprenderertask.cpp

```diff
@@ -190,6 +194,21 @@ bool QgsMapRendererTask::run()
       QRectF rect( 0, 0, img.width(), img.height() );
       pp.drawImage( rect, img, rect );
       pp.end();
+
+      if ( mSaveWorldFile )
+      {
+        CPLSetConfigOption( "GDAL_PDF_DPI", QString::number( mMapSettings.outputDpi() ).toLocal8Bit().constData() );
+        gdal::dataset_unique_ptr outputDS( GDALOpen( mFileName.toLocal8Bit().constData(), GA_Update ) );
+        if ( outputDS )
+        {
+          double a, b, c, d, e, f;
+          QgsMapSettingsUtils::worldFileParameters( mMapSettings, a, b, c, d, e, f );
```

Comment:
The use of worldFileParameters() (provided that this method does return values for a ESRI .wld !) is not really appropriate to set a GDAL geotransform. You did the coefficient order change right, but there's also a half-pixel difference in the origin between worldfile and GDAL geotransform conventions. See https://github.com/OSGeo/gdal/blob/master/gdal/gcore/gdal_misc.cpp#L1841 for the relationships between both